TACC-Cloud/geoapi
# Task/DES-2198: do not re-attempt to import files that previously failed

## Overview: ##

Do not reattempt to import files that already failed.

## PR Status: ##


## Related Jira tickets: ##

* [DES-2198](https://jira.tacc.utexas.edu/browse/DES-2198)

## Summary of Changes: ##

* Note which files failed and do not repeat
* Ensure debug logging level when in DEBUG mode
* Log elapsed time of refresh_observable_projects

## Testing Steps: ##
1. Update geoapi/celery_app.py so you can see project refresh:
```
import datetime
 app.conf.beat_schedule = {
     'refresh_observable_projects': {
         'task': 'geoapi.tasks.external_data.refresh_observable_projects',
        'schedule': datetime.timedelta(seconds=90)
     }
 }
```
(might need to do a `rm geoapi/celerybeat.pid` and `rm geoapi/celerybeat-schedule' before running `docker-compose up`)

2. Then, create a new synced Map of "Observable-Project-Test-Example" which has two good files and one bad file.
3. Note that the failed file import is noted and that we no longer attempt to import the bad file when scraping the folder.

Also, run worker-related unit tests: `APP_ENV=testing pytest -m worker`

## Notes: ##
